src/strategies/conv-finance/index.ts

```diff
@@ -0,0 +1,343 @@
+    if (stakingPool.version === '2') {
+      stakingPool.pools.forEach((poolInfo: STAKING_POOL_INFO) => {
+        stakingPoolsMultiCalls.push(
+          multicall(
+            network,
+            provider,
+            stakingPoolsV2ContractAbi,
+            [
+              [stakingPool.address, 'poolInfos', [poolInfo.poolId]], // Get pool token
+              ...addresses.map((userAddress: any) => [
+                stakingPool.address,
+                'userData',
+                [poolInfo.poolId, userAddress]
+              ]),
+              ...addresses.map((userAddress: any) => [
+                stakingPool.address,
+                'getReward',
+                [poolInfo.poolId, userAddress, poolInfo.rewarderIdx]
+              ])
+            ],
+            { blockTag }
+          )
+        );
+      });
+    } else {
+      stakingPool.pools.forEach((poolInfo: STAKING_POOL_INFO) => {
+        stakingPoolsMultiCalls.push(
+          multicall(
+            network,
+            provider,
+            stakingPoolsV1ContractAbi,
+            [
+              [stakingPool.address, 'poolInfos', [poolInfo.poolId]], // Get pool token
+              ...addresses.map((userAddress: any) => [
+                stakingPool.address,
+                'userData',
+                [poolInfo.poolId, userAddress]
+              ]),
+              ...addresses.map((userAddress: any) => [
+                stakingPool.address,
+                'getReward',
+                [poolInfo.poolId, userAddress]
+              ])
+            ],
+            { blockTag }
+          )
+        );
+      });
+    }
+  });
+
+  const stakingPoolRewarderMultiCalls: any[] = [];
+
+  options.rewarder.forEach((rewarder: REWARDER) => {
+    //  Staking pool rewarder version:
+    // 1: Old rewarder
+    // 2. New rewarder
+    if (rewarder.version === '2') {
+      rewarder.poolIds.forEach((id: number) => {
+        stakingPoolRewarderMultiCalls.push(
+          multicall(
+            network,
+            provider,
+            rewarderV2ContractAbi,
+            [
+              ...addresses.map((userAddress: any) => [
+                rewarder.address,
+                'calculateTotalReward',
+                [userAddress, id]
+              ])
+            ],
+            { blockTag }
+          )
+        );
+      });
+    } else {
+      rewarder.poolIds.forEach((id: number) => {
+        stakingPoolRewarderMultiCalls.push(
+          multicall(
+            network,
+            provider,
+            rewarderV1ContractAbi,
+            [
+              ...addresses.map((userAddress: any) => [
+                rewarder.address,
+                'vestingSchedules',
+                [userAddress, id]
+              ])
+            ],
+            { blockTag }
+          )
+        );
+      });
+    }
+  });
+
+  let res = await Promise.all([
+    ...lpMultiCalls,
+    ...stakingPoolsMultiCalls,
+    ...stakingPoolRewarderMultiCalls
+  ]);
```

Comment:
Hi @eric-convergence Can you add a limit for `options.rewarder.poolIds` and `options.stakingPools` like this it may send many requests if the length of these is more , there is a limit of 5 requests from a strategy https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

Reply:
Tried my best to add limit in the new commit